incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -3109,23 +3058,26 @@ private void onSessionReset(
                 final OctetsFW extension = reset.extension();
                 final MqttResetExFW mqttResetEx = extension.get(mqttResetExRO::tryWrap);
 
-                String16FW serverReference = mqttResetEx.serverReference();
-                byte reasonCode = SUCCESS;
-                if (serverReference != null && serverReference.length() != 0)
-                {
-                    reasonCode = SERVER_MOVED;
-                }
-                if (!connected)
-                {
-                    doCancelConnectTimeout();
 
-                    doEncodeConnack(traceId, authorization, reasonCode, assignedClientId, false, serverReference);
-                }
-                else
+
+                if (mqttResetEx != null)
                 {
-                    doEncodeDisconnect(traceId, authorization, reasonCode, serverReference);
-                }
+                    String16FW serverReference = mqttResetEx.serverReference();
+                    boolean serverReferenceExists = serverReference != null && serverReference.length() != 0;
```

Comment:
Why do we have a special value check for empty server reference?
Shouldn't it just be `!= null` and use whatever is given, since we're not validating the value in any other way.